src/StaticWebAssetsSdk/Tasks/DefineStaticWebAssets.Cache.cs

```diff
@@ -0,0 +1,217 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.StaticWebAssets.Tasks.Utils;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using static Microsoft.AspNetCore.StaticWebAssets.Tasks.FingerprintPatternMatcher;
+
+namespace Microsoft.AspNetCore.StaticWebAssets.Tasks;
+
+public partial class DefineStaticWebAssets : Task
+{
+    private DefineStaticWebAssetsCache GetOrCreateAssetsCache()
+    {
+        var memoryStream = new MemoryStream();
+        var properties = new[] {
+            SourceId, SourceType, BasePath, ContentRoot, RelativePathPattern, RelativePathFilter,
+            AssetKind, AssetMode, AssetRole, AssetMergeSource, AssetMergeBehavior, RelatedAsset,
+            AssetTraitName, AssetTraitValue, CopyToOutputDirectory, CopyToPublishDirectory,
+            FingerprintCandidates.ToString()
+        };
+        var propertiesHash = HashingUtils.ComputeHash(memoryStream, properties);
+
+        var patternMetadata = new[] { nameof(FingerprintPattern.Pattern), nameof(FingerprintPattern.Expression) };
+        var fingerprintPatternsHash = HashingUtils.ComputeHash(memoryStream, FingerprintPatterns ?? [], patternMetadata);
+
+        var propertyOverridesHash = HashingUtils.ComputeHash(memoryStream, PropertyOverrides, nameof(ITaskItem.GetMetadata));
```

Comment:
Is it valid to check if `CacheManifestPath` is null at the start of this method and short-circuit some of this work? Or maybe there's a way to structure things such that the work done to construct the cache doesn't happen if we're never going to output the cache anyway? I noticed that there are still quite a few invocations of this task that don't define a cache manifest path. Do we know if there's a measurable negative performance impact there?